### PR TITLE
Fix: Event Nudge Pref Link 

### PR DESCRIPTION
**Pull request by abdullai-t**
#### works with
-  Portal [PR](https://github.com/massenergize/frontend-portal/pull/1265)
- Admin [PR](https://github.com/massenergize/frontend-admin/pull/1092)

#### Changes
- [x] Added login_method to the args of the login route
- [x] Updated the URL for changing communication preferences in event nudge to include user data and login method

### CODE REVIEW

File: src/task_queue/events_nudge/cadmin_events_nudge.py [hunk 1]
```diff
@@ -205,7 +213,8 @@ def send_events_nudge():
 
 def send_events_report(name, email, event_list):
     try:
-        change_preference_link = ADMIN_URL_ROOT+"/admin/profile/preferences"
+        cred = encode_data({"email": email})
```
Comment (BradHN1) [inline src/task_queue/events_nudge/cadmin_events_nudge.py:216]: Can you move this duplicated code to a utility routine for example add_credentials()?  I think we will be using it a number of links.

File: src/task_queue/events_nudge/cadmin_events_nudge.py [hunk 2]
```diff
@@ -205,7 +205,8 @@ def send_events_nudge():
 
 def send_events_report(name, email, event_list):
     try:
-        change_preference_link = ADMIN_URL_ROOT+"/admin/profile/preferences"
+        cred = encode_data_for_URL({"email": email})
```
Comment (BradHN1) [inline src/task_queue/events_nudge/cadmin_events_nudge.py:208]: should this also include the login_method (or could that login_method always be added if an email is specified)?

Comment (abdullai-t) [reply src/task_queue/events_nudge/cadmin_events_nudge.py:208]: the login_method can be included.

Comment (BradHN1) [reply src/task_queue/events_nudge/cadmin_events_nudge.py:208]: If you have the utility routine add that may simplify the usage

**Review by BradHN1 — APPROVED** [review]
LGTM

**Pull request merged**